launcher/modplatform/modrinth/ModrinthAPI.cpp

```diff
@@ -120,3 +120,41 @@ QList<ResourceAPI::SortingMethod> ModrinthAPI::getSortingMethods() const
              { 4, "newest", QObject::tr("Sort by Newest") },
              { 5, "updated", QObject::tr("Sort by Last Updated") } };
 }
+
+Task::Ptr ModrinthAPI::getModCategories(std::shared_ptr<QByteArray> response)
+{
+    auto netJob = makeShared<NetJob>(QString("Modrinth::GetCategories"), APPLICATION->network());
+    netJob->addNetAction(Net::ApiDownload::makeByteArray(QUrl(BuildConfig.MODRINTH_PROD_URL + "/tag/category"), response));
+    QObject::connect(netJob.get(), &Task::failed, [](QString msg) { qDebug() << "Modrinth failed to get categories:" << msg; });
+    return netJob;
+}
+
+QList<ModPlatform::Category> ModrinthAPI::loadModCategories(std::shared_ptr<QByteArray> response)
+{
+    QList<ModPlatform::Category> categories;
+    QJsonParseError parse_error{};
+    QJsonDocument doc = QJsonDocument::fromJson(*response, &parse_error);
+    if (parse_error.error != QJsonParseError::NoError) {
+        qWarning() << "Error while parsing JSON response from categories at " << parse_error.offset
+                   << " reason: " << parse_error.errorString();
+        qWarning() << *response;
+        return categories;
+    }
+
+    try {
+        auto arr = Json::requireArray(doc);
+
+        for (auto val : arr) {
+            auto cat = Json::requireObject(val);
+            auto name = Json::requireString(cat, "name");
```

Comment:
these names should be converted to title case (or at least the first letter should be uppercase) to look better in the ui. modrinth's api returns them all in lowercase

Reply:
Done